src/QMCWaveFunctions/Fermion/syclSolverInverter.hpp

```diff
@@ -0,0 +1,135 @@
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_SYCL_MKL_SOLVERINVERTOR_H
+#define QMCPLUSPLUS_SYCL_MKL_SOLVERINVERTOR_H
+
+#include "OhmmsPETE/OhmmsVector.h"
+#include "OhmmsPETE/OhmmsMatrix.h"
+#include "SYCL/SYCLallocator.hpp"
+#include "SYCL/syclBLAS.hpp"
+#include "SYCL/syclSolver.hpp"
+#include "QMCWaveFunctions/detail/SYCL/sycl_determinant_helper.hpp"
+
+namespace qmcplusplus
+{
+/** implements matrix inversion via cuSolverDN
+ * @tparam T_FP high precision for matrix inversion, T_FP >= T
+ */
+template<typename T_FP>
+class syclSolverInverter
+{
+  /// scratch memory for cusolverDN
+  Matrix<T_FP, SYCLAllocator<T_FP>> Mat1_gpu;
+  /// pivot array + info
+  Vector<std::int64_t, SYCLAllocator<std::int64_t>> ipiv;
+  /// workspace
+  Vector<T_FP, SYCLAllocator<T_FP>> workspace;
+  std::int64_t getrf_ws = 0;
+  std::int64_t getri_ws = 0;
+
+  /** resize the internal storage
+   * @param norb number of electrons/orbitals
+   * @param delay, maximum delay 0<delay<=norb
+   */
+  inline void resize(int norb, sycl::queue& m_queue)
+  {
+    if (Mat1_gpu.rows() != norb)
+    {
+      Mat1_gpu.resize(norb, norb);
+      ipiv.resize(norb);
+      getrf_ws = syclSolver::getrf_scratchpad_size<T_FP>(m_queue, norb, norb, norb);
+      getri_ws = syclSolver::getri_scratchpad_size<T_FP>(m_queue, norb, norb);
+      workspace.resize(std::max(getrf_ws, getri_ws));
+    }
+  }
+
+public:
+  /** compute the inverse of the transpose of matrix A and its determinant value in log
+   * when T_FP and TMAT are the same
+   * @tparam TREAL real type
+   */
+  template<typename TMAT, typename TREAL, typename = std::enable_if_t<std::is_same<TMAT, T_FP>::value>>
+  std::enable_if_t<std::is_same<TMAT, T_FP>::value> invert_transpose(const Matrix<TMAT>& logdetT,
+                                                                     Matrix<TMAT>& Ainv,
+                                                                     Matrix<TMAT, SYCLAllocator<TMAT>>& Ainv_gpu,
+                                                                     std::complex<TREAL>& log_value,
+                                                                     sycl::queue& m_queue)
+  {
+    const int norb = logdetT.rows();
+    resize(norb, m_queue);
+
+    auto c_event = m_queue.memcpy(Mat1_gpu.data(), logdetT.data(), logdetT.size() * sizeof(TMAT));
+    auto t_event = syclBLAS::transpose(m_queue, Mat1_gpu.data(), norb, Mat1_gpu.cols(), Ainv_gpu.data(), norb,
+                                       Ainv_gpu.cols(), {c_event});
+    try
+    {
+      syclSolver::getrf(m_queue, norb, norb, Ainv_gpu.data(), norb, ipiv.data(), workspace.data(), getrf_ws,
+                        {t_event}).wait();
+    }
+    catch (cl::sycl::exception const& ex)
+    {
+      std::cout << "\t\tCaught synchronous SYCL exception during getrf:\n"
+                << ex.what() << "  status: " << ex.code() << std::endl;
+      abort();
```

Comment:
Don't do abort. throw an std::runtime_error with the cout message.

Reply:
OK